tests/e2e/lib/apps.go

```diff
@@ -65,7 +66,24 @@ func InstallApplication(ocClient client.Client, file string) error {
 		resource.SetLabels(labels)
 		err = ocClient.Create(context.Background(), &resource)
 		if apierrors.IsAlreadyExists(err) {
-			continue
+			// if spec has changed for following kinds, update the resource
+			clusterResource := unstructured.Unstructured{
+				Object: resource.Object,
+			}
+			err = ocClient.Get(context.Background(), types.NamespacedName{Name: resource.GetName(), Namespace: resource.GetNamespace()}, &clusterResource)
+			if err != nil {
+				return err
+			}
```

Comment:
Shouldn't we do a resource diff check first ? like `DeepEquals` ?

Reply:
Could do that and add exceptions for metadata and status

Reply:
yeah lets do that, it would reduce the unnecessary update calls.

Reply:
+1

Reply:
resolved in b1edea0aa1f49c83deed47de2cbc00cac9808778